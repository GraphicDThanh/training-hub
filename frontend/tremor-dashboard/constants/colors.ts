type ColorChannelsChartType = {
  [key: number]: string;
  1: string;
  2: string;
  3: string;
  4: string;
};

export const COLOR_CHANNELS_CHART: ColorChannelsChartType = {
  1: "blue-500",
  2: "red-500",
  3: "primary",
  4: "gray-500",
};
