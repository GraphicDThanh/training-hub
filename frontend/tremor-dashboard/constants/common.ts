// Types
import { OptionType } from "@/types";

export const EXCEPT_KEYS = {
  POSITIVE_INTEGER: ["e", "E", "+", "-", "."],
  POSITIVE_DOUBLE: ["e", "E", "+", "-"],
};

export enum DIRECTION {
  ASC = "asc",
  DESC = "desc",
}

// Mailto Support
export const MAILTO_SUPPORT = "support@tremor-dashboard.com";

export const REMEMBER_ME_COOKIES_KEY = "remember-me";
export const UID_KEY = "uid";
export const ROLE = "role";
export const NAME = "name";
export const AVATAR = "avatar";
export const AUTH_SESSION_COOKIES_KEY = "authjs.session-token";

export const ORDER_LIST_OPTIONS: OptionType[] = [
  { option: "Paid", value: "0" },
  { option: "Refunded", value: "2" },
  { option: "Canceled", value: "1" },
];

export const PRODUCT_LIST_OPTIONS: OptionType[] = [
  { option: "Yes", value: "true" },
  { option: "No", value: "false" },
];

export const FORMAT_DATE = {
  YEAR_MONTH_DATE_TIME: "YYYY-MM-DDTHH:mm:ss.SSS",
  MONTH_DAY_YEAR_TIME: "MMM D, YYYY h:mm A",
};

export const SEPARATOR = {
  COMMAS: ",",
  DOTS: ".",
};

export const CDN_KEY = process.env.NEXT_PUBLIC_CDN_KEY;

export const DRAG_ZONE = {
  DEFAULT: {
    TEXT: "Drag and drop your file here to begin uploading.",
    STYLE_INPUT: "border-gray-300",
    STYLE_TEXT: "text-gray-500 dark:text-gray-400",
  },
  ON_DRAG: {
    TEXT: "Release the mouse click to upload the photo",
    STYLE_INPUT: "outline outline-2 outline-[#ebeff4]",
    STYLE_TEXT: "text-gray-500 dark:text-gray-400",
  },
  ON_WRONG_FORMAT: {
    TEXT: "Invalid image file type. Please try uploading again.",
    STYLE_INPUT: "border-red-500",
    STYLE_TEXT: "text-red-500 dark:text-red-500",
  },
};

export const TRANSACTION_CLASS = {
  INCREASE: {
    buttonClasses: "text-few dark:text-few dark:hove:text-few border-few",
    textClasses: "font-semibold text-greener dark:text-few",
  },
  DECREASE: {
    buttonClasses:
      "text-attention dark:text-attention dark:hover:text-attention border-attention",
    textClasses: "font-semibold text-blood dark:text-pink",
  },
  PENDING: {
    buttonClasses:
      "text-primary dark:text-primary dark:hover:text-primary border-primary",
    textClasses:
      "font-semibold text-primary dark:text-dark-tremor-content-pending",
  },
};

export const enum VARIANT_BUTTON {
  LIGHT = "light",
  LIGHT_CARD = "lightCard",
  LIGHT_CENTER = "lightCenter",
  LIGHT_STATUS = "lightStatus",
  SECONDARY = "secondary",
  SECONDARY_SHADOW = "secondaryShadow",
  PRIMARY = "primary",
  PRIMARY_CENTER = "primaryCenter",
  SURFACE = "surface",
  DARK = "dark",
  ACTIVE = "active",
}

export const NO_IMAGE = "No Image";

export const RESULT_NOT_FOUND = "Result Not Found";

export const AMOUNT_LIMIT = 20000;
export const PRICE_LIMIT = 20000;
export const QUANTITY_LIMIT = 10000;
export const WEIGHT_LIMIT = 2000;

export const PREFIX_PHONE_NUMBER = "+1";

export const STORAGES = {
  IS_DARK_THEME: "isDarkTheme",
};

export const THEMES = {
  LIGHT: "light",
  DARK: "dark",
};

export const LOGO_TITLE = "Tremor Dashboard PRO";
