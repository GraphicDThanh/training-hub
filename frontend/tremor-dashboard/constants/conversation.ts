export const CONVERSATION_TABLES = {
  GROUPS: "groups",
  CHATS: "chats",
};

export const CONVERSATION_ORDERS = {
  CREATED_AT: "createdAt",
  UPDATED_AT: "updatedAt",
};

export const CONVERSATION_QUERYS = {
  USER_IDS: "userIds",
};

export enum CONVERSATION_STATUS {
  INITIAL = "initial",
  FETCHED = "fetched",
}

export const ADMIN_ID = parseInt(process.env.NEXT_PUBLIC_ADMIN_ID!);

export enum DOCUMENT_CHANGE_TYPE {
  ADDED = "added",
  MODIFIED = "modified",
  REMOVED = "removed",
}

export enum WHERE_FILTER_OP {
  ARRAY_CONTAINS = "array-contains",
}
