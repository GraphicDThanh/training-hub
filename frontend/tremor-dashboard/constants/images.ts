export const NOT_FOUND_IMAGE = "/images/no-image-square.webp";
export const LOGO_IMAGE = "/assets/images/sidebar-logo.webp";
export const PLACEHOLDER_IMAGE = "/images/image-regular.svg";

export const AVATAR_IMAGE = {
  XS: "/images/avatar/avatar-xs.webp",
  SM: "/images/avatar/avatar-sm.webp",
  MD: "/images/avatar/avatar-md.webp",
  LG: "/images/avatar/avatar-lg.webp",
};

export const METADATA_IMAGE = {
  HOME_PREVIEW: "/assets/images/home-preview.webp",
  MASTERCARD: "/assets/images/mastercard.webp",
};

export const PROJECT_IMAGE = "/assets/images/profile-project.webp";
