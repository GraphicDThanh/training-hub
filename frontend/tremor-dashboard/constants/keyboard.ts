export const KEYBOARD_KEYS = {
  BACKSPACE: "Backspace",
  DIGIT_1: "1",
};
