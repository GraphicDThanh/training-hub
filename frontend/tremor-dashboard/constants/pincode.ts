export const PIN_CODE_DEFAULT = "*";
export const PIN_CODE_LENGTH = 6;
export const PIN_CODE_SKIP_SET_KEY = "skipSetupPinCode";
