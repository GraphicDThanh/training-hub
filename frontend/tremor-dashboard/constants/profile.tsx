import { PlatformSetting } from "@/types";

export const SETTING_FIELDS_ACCOUNT: PlatformSetting[] = [
  { label: "Email me when someone follows me", field: "emailMentions" },
  {
    label: "Email me when someone answers on my post",
    field: "emailFollowing",
  },
  { label: "Email me when someone mentions me", field: "emailAnswerPost" },
];

export const SETTING_FIELDS_APPLICATION: PlatformSetting[] = [
  { label: "New launches and projects", field: "newLaunchesProject" },
  { label: "Monthly product updates", field: "monthlyProductUpdate" },
  { label: "Subscribe to newsletter", field: "subscribeToNewsletter" },
];

export const SETTING_DATA_ACCOUNT = {
  emailMentions: false,
  emailFollowing: false,
  emailAnswerPost: false,
};

export const SETTING_DATA_APPLICATION = {
  newLaunchesProject: false,
  monthlyProductUpdate: false,
  subscribeToNewsletter: false,
};

// Social links
export const SOCIAL_LINKS = {
  facebook: "https://www.facebook.com/",
  twitter: "https://twitter.com/",
  instagram: "https://instagram.com",
};
