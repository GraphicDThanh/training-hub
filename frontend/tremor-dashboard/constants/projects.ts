import { ImageProject, User } from "@/types";

export const TOOL_ICON_OPTION: ImageProject[] = [
  {
    name: "SlackBot",
    id: 0,
  },
  {
    name: "Spotify",
    id: 1,
  },
  {
    name: "Yahoo",
    id: 2,
  },
  {
    name: "Zalo",
    id: 3,
  },
  {
    name: "Twitter",
    id: 4,
  },
];

export const PARTICIPANTS_PROJECTS = [
  {
    id: 0,
    name: "Jessica",
    avatar:
      "https://demos.creative-tim.com/nextjs-material-das…oard-pro//_next/static/media/logo-xd.4ef94f7b.svg",
  },
  {
    id: 9,
    name: "Kali",
    avatar:
      "https://demos.creative-tim.com/nextjs-material-das…oard-pro//_next/static/media/logo-xd.4ef94f7b.svg",
  },
  {
    id: 5,
    name: "Henry ",
    avatar:
      "https://demos.creative-tim.com/nextjs-material-das…oard-pro//_next/static/media/logo-xd.4ef94f7b.svg",
  },
  {
    id: 6,
    name: "John",
    avatar:
      "https://demos.creative-tim.com/nextjs-material-das…oard-pro//_next/static/media/logo-xd.4ef94f7b.svg",
  },
  {
    id: 4,
    name: "June",
    avatar:
      "https://demos.creative-tim.com/nextjs-material-das…oard-pro//_next/static/media/logo-xd.4ef94f7b.svg",
  },
] as User[];

// Max should is 10 for all screen.
export const PARTICIPANTS_NUMBER_PROJECT = 10;
