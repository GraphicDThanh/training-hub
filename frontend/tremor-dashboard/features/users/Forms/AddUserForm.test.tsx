// Libs
import { fireEvent, render, waitFor } from "@testing-library/react";
import React from "react";

// Components
import { AddUserForm } from ".";

// Constants
import { ADMIN_ID } from "@/constants";

// Mocks
import { MOCK_USERS } from "@/mocks";

jest.mock("next/navigation", () => {
  const actual = jest.mock("next/navigation");
  return {
    ...actual,
    useRouter: jest.fn(() => ({
      push: jest.fn(),
      replace: jest.fn(),
    })),
    useSearchParams: jest.fn(() => ({ get: jest.fn(() => "mockPage") })),
    usePathname: jest.fn(),
  };
});

jest.mock("next/cache", () => ({
  revalidatePath: jest.fn(),
}));

// Common setup function
const setup = (props?: { userId?: number }) =>
  render(<AddUserForm userId={props?.userId ?? ADMIN_ID} />);

describe("AddUserForm", () => {
  const { name, email, password, bio, twitterUrl, facebookUrl, instagramUrl } =
    MOCK_USERS[0];

  it("should render successfully in User Info Form", async () => {
    const { container, getByLabelText, getByTestId, getAllByTestId } = setup();

    // FORM 1
    fireEvent.change(getByLabelText("Name"), { target: { value: name } });
    fireEvent.change(getByLabelText("Email"), {
      target: { value: email },
    });
    fireEvent.click(getByTestId("year").firstElementChild as Element);
    fireEvent.click(getAllByTestId("select-option")[1]);
    fireEvent.click(getByTestId("month").firstElementChild as Element);
    fireEvent.click(getAllByTestId("select-option")[2]);
    fireEvent.click(getByTestId("date").firstElementChild as Element);
    fireEvent.click(getAllByTestId("select-option")[3]);

    fireEvent.change(getByLabelText("Password"), {
      target: { value: password },
    });
    fireEvent.change(getByLabelText("Confirm Password"), {
      target: { value: password },
    });

    expect((getByLabelText("Name") as HTMLInputElement).value).toBe(name);
    expect((getByLabelText("Email") as HTMLInputElement).value).toBe(email);
    expect(
      getByTestId("year").firstElementChild?.firstElementChild?.textContent,
    ).toBe("1901");
    expect(
      getByTestId("month").firstElementChild?.firstElementChild?.textContent,
    ).toBe("March");
    expect(
      getByTestId("date").firstElementChild?.firstElementChild?.textContent,
    ).toBe("4");
    expect((getByLabelText("Password") as HTMLInputElement).value).toBe(
      password,
    );
    expect((getByLabelText("Confirm Password") as HTMLInputElement).value).toBe(
      password,
    );

    fireEvent.submit(container.querySelector("#user-info-form") as Element);

    await waitFor(() =>
      expect(
        getByTestId("step-content-2").querySelectorAll("span")[1].className,
      ).toBe("stepper-dot-active"),
    );
  });

  it("should render successfully in Profile Form", async () => {
    const { container, getByLabelText, getByTestId, getAllByTestId } = setup();

    // FORM 1
    fireEvent.change(getByLabelText("Name"), { target: { value: name } });
    fireEvent.change(getByLabelText("Email"), {
      target: { value: email },
    });
    fireEvent.click(getByTestId("year").firstElementChild as Element);
    fireEvent.click(getAllByTestId("select-option")[1]);
    fireEvent.click(getByTestId("month").firstElementChild as Element);
    fireEvent.click(getAllByTestId("select-option")[2]);
    fireEvent.click(getByTestId("date").firstElementChild as Element);
    fireEvent.click(getAllByTestId("select-option")[3]);
    fireEvent.change(getByLabelText("Password"), {
      target: { value: password },
    });
    fireEvent.change(getByLabelText("Confirm Password"), {
      target: { value: password },
    });

    expect((getByLabelText("Name") as HTMLInputElement).value).toBe(name);
    expect((getByLabelText("Email") as HTMLInputElement).value).toBe(email);
    expect(
      getByTestId("year").firstElementChild?.firstElementChild?.textContent,
    ).toBe("1901");
    expect(
      getByTestId("month").firstElementChild?.firstElementChild?.textContent,
    ).toBe("March");
    expect(
      getByTestId("date").firstElementChild?.firstElementChild?.textContent,
    ).toBe("4");
    expect((getByLabelText("Password") as HTMLInputElement).value).toBe(
      password,
    );
    expect((getByLabelText("Confirm Password") as HTMLInputElement).value).toBe(
      password,
    );

    fireEvent.submit(container.querySelector("#user-info-form") as Element);

    await waitFor(() =>
      getByTestId("step-content-2").querySelector(
        "span[class='stepper-dot-active']",
      ),
    );

    // FORM 2
    await waitFor(() => container.querySelector("#profile-form"));

    fireEvent.change(container.querySelector(".ql-editor") as Element, {
      target: { textContent: bio },
    });

    fireEvent.submit(container.querySelector("#profile-form") as Element);

    await waitFor(() =>
      expect(
        getByTestId("last-step-content").querySelectorAll("span")[1].className,
      ).toBe("stepper-dot-active"),
    );
  });

  it("should render successfully in Social Form", async () => {
    const { container, getByLabelText, getByTestId, getAllByTestId } = setup();

    const responseData = { success: true };
    const mockFetch = jest.fn();
    const mockResponse = {
      ok: true,
      json: mockFetch.mockResolvedValue(responseData),
    };

    // Mock fetch
    global.fetch = mockFetch.mockResolvedValue(mockResponse);

    // FORM 1:
    fireEvent.change(getByLabelText("Name"), { target: { value: name } });
    fireEvent.change(getByLabelText("Email"), {
      target: { value: email },
    });
    fireEvent.click(getByTestId("year").firstElementChild as Element);
    fireEvent.click(getAllByTestId("select-option")[1]);
    fireEvent.click(getByTestId("month").firstElementChild as Element);
    fireEvent.click(getAllByTestId("select-option")[2]);
    fireEvent.click(getByTestId("date").firstElementChild as Element);
    fireEvent.click(getAllByTestId("select-option")[3]);
    fireEvent.change(getByLabelText("Password"), {
      target: { value: password },
    });
    fireEvent.change(getByLabelText("Confirm Password"), {
      target: { value: password },
    });

    expect((getByLabelText("Name") as HTMLInputElement).value).toBe(name);
    expect((getByLabelText("Email") as HTMLInputElement).value).toBe(email);
    expect(
      getByTestId("year").firstElementChild?.firstElementChild?.textContent,
    ).toBe("1901");
    expect(
      getByTestId("month").firstElementChild?.firstElementChild?.textContent,
    ).toBe("March");
    expect(
      getByTestId("date").firstElementChild?.firstElementChild?.textContent,
    ).toBe("4");
    expect((getByLabelText("Password") as HTMLInputElement).value).toBe(
      password,
    );
    expect((getByLabelText("Confirm Password") as HTMLInputElement).value).toBe(
      password,
    );

    fireEvent.submit(container.querySelector("#user-info-form") as Element);

    await waitFor(() =>
      getByTestId("step-content-2").querySelector(
        "span[class='stepper-dot-active']",
      ),
    );

    // FORM 2:
    await waitFor(() => container.querySelector("#profile-form"));

    fireEvent.change(container.querySelector(".ql-editor") as Element, {
      target: { textContent: bio },
    });

    fireEvent.submit(container.querySelector("#profile-form") as Element);

    await waitFor(() =>
      getByTestId("last-step-content").querySelector(
        "span[class='stepper-dot-active']",
      ),
    );

    // FORM 3
    await waitFor(() => container.querySelector("#social-form"));

    fireEvent.change(getByLabelText("Twitter Handle"), {
      target: { value: twitterUrl },
    });
    fireEvent.change(getByLabelText("Facebook Account"), {
      target: { value: facebookUrl },
    });
    fireEvent.change(getByLabelText("Instagram Account"), {
      target: { value: instagramUrl },
    });

    fireEvent.submit(container.querySelector("#social-form") as Element);

    expect(container).toMatchSnapshot();
    await waitFor(() => expect(mockFetch).toHaveBeenCalled());
  });

  it("should render success toast when message null", async () => {
    const { container, getByLabelText, getByTestId, getAllByTestId } = setup();

    const mockFetch = jest.fn();
    const mockResponse = {
      ok: false,
      json: () => Promise.resolve({ message: "Error" }),
    };

    // Mock fetch
    global.fetch = mockFetch.mockResolvedValue(mockResponse);

    // FORM 1:
    fireEvent.change(getByLabelText("Name"), { target: { value: name } });
    fireEvent.change(getByLabelText("Email"), {
      target: { value: email },
    });

    fireEvent.click(getByTestId("year").firstElementChild as Element);
    fireEvent.click(getAllByTestId("select-option")[1]);
    fireEvent.click(getByTestId("month").firstElementChild as Element);
    fireEvent.click(getAllByTestId("select-option")[2]);
    fireEvent.click(getByTestId("date").firstElementChild as Element);
    fireEvent.click(getAllByTestId("select-option")[3]);
    fireEvent.change(getByLabelText("Password"), {
      target: { value: password },
    });
    fireEvent.change(getByLabelText("Confirm Password"), {
      target: { value: password },
    });

    expect((getByLabelText("Name") as HTMLInputElement).value).toBe(name);
    expect((getByLabelText("Email") as HTMLInputElement).value).toBe(email);
    expect(
      getByTestId("year").firstElementChild?.firstElementChild?.textContent,
    ).toBe("1901");
    expect(
      getByTestId("month").firstElementChild?.firstElementChild?.textContent,
    ).toBe("March");
    expect(
      getByTestId("date").firstElementChild?.firstElementChild?.textContent,
    ).toBe("4");
    expect((getByLabelText("Password") as HTMLInputElement).value).toBe(
      password,
    );
    expect((getByLabelText("Confirm Password") as HTMLInputElement).value).toBe(
      password,
    );

    fireEvent.submit(container.querySelector("#user-info-form") as Element);

    await waitFor(() =>
      getByTestId("step-content-2").querySelector(
        "span[class='stepper-dot-active']",
      ),
    );

    // FORM 2:
    await waitFor(() => container.querySelector("#profile-form"));

    fireEvent.change(container.querySelector(".ql-editor") as Element, {
      target: { textContent: bio },
    });

    fireEvent.submit(container.querySelector("#profile-form") as Element);

    await waitFor(() =>
      getByTestId("last-step-content").querySelector(
        "span[class='stepper-dot-active']",
      ),
    );

    // FORM 3
    await waitFor(() => container.querySelector("#social-form"));

    fireEvent.change(getByLabelText("Twitter Handle"), {
      target: { value: twitterUrl },
    });
    fireEvent.change(getByLabelText("Facebook Account"), {
      target: { value: facebookUrl },
    });
    fireEvent.change(getByLabelText("Instagram Account"), {
      target: { value: instagramUrl },
    });

    fireEvent.submit(container.querySelector("#social-form") as Element);
    await waitFor(() => expect(mockFetch).toHaveBeenCalled());
  });

  it("should calls back step", async () => {
    const { container, getByLabelText, getByTestId, getAllByTestId } = setup();

    // FORM 1
    fireEvent.change(getByLabelText("Name"), { target: { value: name } });
    fireEvent.change(getByLabelText("Email"), {
      target: { value: email },
    });
    fireEvent.click(getByTestId("year").firstElementChild as Element);
    fireEvent.click(getAllByTestId("select-option")[1]);
    fireEvent.click(getByTestId("month").firstElementChild as Element);
    fireEvent.click(getAllByTestId("select-option")[2]);
    fireEvent.click(getByTestId("date").firstElementChild as Element);
    fireEvent.click(getAllByTestId("select-option")[3]);
    fireEvent.change(getByLabelText("Password"), {
      target: { value: password },
    });
    fireEvent.change(getByLabelText("Confirm Password"), {
      target: { value: password },
    });

    expect((getByLabelText("Name") as HTMLInputElement).value).toBe(name);
    expect((getByLabelText("Email") as HTMLInputElement).value).toBe(email);
    expect(
      getByTestId("year").firstElementChild?.firstElementChild?.textContent,
    ).toBe("1901");
    expect(
      getByTestId("month").firstElementChild?.firstElementChild?.textContent,
    ).toBe("March");
    expect(
      getByTestId("date").firstElementChild?.firstElementChild?.textContent,
    ).toBe("4");
    expect((getByLabelText("Password") as HTMLInputElement).value).toBe(
      password,
    );
    expect((getByLabelText("Confirm Password") as HTMLInputElement).value).toBe(
      password,
    );

    fireEvent.submit(container.querySelector("#user-info-form") as Element);

    await waitFor(() =>
      getByTestId("step-content-2").querySelector(
        "span[class='stepper-dot-active']",
      ),
    );

    // FORM 2
    await waitFor(() => container.querySelector("#profile-form"));

    fireEvent.click(getByTestId("back-button"));

    await waitFor(() =>
      expect(
        getByTestId("step-content-2").querySelectorAll("span")[1].className,
      ).toBe("stepper-dot"),
    );
  });
});
