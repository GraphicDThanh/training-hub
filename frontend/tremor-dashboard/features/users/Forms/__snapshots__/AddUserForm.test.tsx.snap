// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddUserForm should render successfully in Social Form 1`] = `
<div>
  <div
    class="flex flex-col items-center relative p-4 bg-white dark:bg-dark-tremor-primary rounded-lg shadow-md mx-auto w-full relative"
  >
    <ul
      class="grid grid-cols-3 pt-6 pb-4 bg-zinc-700 dark:stepper-dark-bg shadow-box-icon-primary rounded-lg absolute -translate-y-10 z-10 w-11/12 dark:bg-gradient-pickled undefined"
      data-testid="stepper"
    >
      <li
        data-testid="step-content-0"
      >
        <div
          class="flex items-center w-full h-4 pl-[50%]"
        >
          <span
            class="stepper-dot-active"
          />
          <span
            class="1234 h-0.5 w-full bg-white"
          />
        </div>
        <p
          class="text-tremor-default dark:text-dark-tremor-content stepper-text-content text-xs text-white mt-2"
        >
          1. User Info
        </p>
      </li>
      <li
        data-testid="step-content-2"
      >
        <div
          class="flex items-center w-full h-4"
        >
          <span
            class="h-0.5 w-[50%] bg-white"
          />
          <span
            class="stepper-dot-active"
          />
          <span
            class="h-0.5 w-[50%] bg-white"
          />
        </div>
        <p
          class="text-tremor-default dark:text-dark-tremor-content stepper-text-content mt-2 text-white text-xs"
        >
          2. Profile
        </p>
      </li>
      <li
        data-testid="last-step-content"
      >
        <div
          class="flex items-center w-full h-4"
        >
          <span
            class="h-0.5 w-[50%] bg-white"
          />
          <span
            class="stepper-dot-active"
          />
        </div>
        <p
          class="stepper-text-content mt-2 text-white"
        >
          3. Social
        </p>
      </li>
    </ul>
    <form
      class="w-full mt-20"
      id="social-form"
    >
      <h6
        class="text-primary dark:text-white font-bold text-xl mb-8"
      >
        Socials
      </h6>
      <div
        class="min-h-20 w-full"
      >
        <div
          class="w-full"
        >
          <label
            class="text-grey text-sm dark:text-lighter"
            for="twitterUrl"
          >
            Twitter Handle
          </label>
          <div
            class="tremor-Flex-root flex w-full flex-row justify-between items-center relative"
          >
            <div
              class="w-full"
            >
              <input
                class="block py-1.5 px-0 w-full text-sm text-primary bg-transparent border-0 border-b-2 border-gray-300 appearance-none dark:text-white dark:border-gray-600 dark:focus:border-gray-500 focus:outline-none focus:ring-0 focus:border-gray-600 peer disabled:cursor-not-allowed disabled:opacity-50 "
                id="twitterUrl"
                name="twitterUrl"
                type="text"
                value="twitter.com"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="min-h-20 w-full"
      >
        <div
          class="w-full"
        >
          <label
            class="text-grey text-sm dark:text-lighter"
            for="facebookUrl"
          >
            Facebook Account
          </label>
          <div
            class="tremor-Flex-root flex w-full flex-row justify-between items-center relative"
          >
            <div
              class="w-full"
            >
              <input
                class="block py-1.5 px-0 w-full text-sm text-primary bg-transparent border-0 border-b-2 border-gray-300 appearance-none dark:text-white dark:border-gray-600 dark:focus:border-gray-500 focus:outline-none focus:ring-0 focus:border-gray-600 peer disabled:cursor-not-allowed disabled:opacity-50 "
                id="facebookUrl"
                name="facebookUrl"
                type="text"
                value="facebook.com"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="min-h-20 w-full"
      >
        <div
          class="w-full"
        >
          <label
            class="text-grey text-sm dark:text-lighter"
            for="instagramUrl"
          >
            Instagram Account
          </label>
          <div
            class="tremor-Flex-root flex w-full flex-row justify-between items-center relative"
          >
            <div
              class="w-full"
            >
              <input
                class="block py-1.5 px-0 w-full text-sm text-primary bg-transparent border-0 border-b-2 border-gray-300 appearance-none dark:text-white dark:border-gray-600 dark:focus:border-gray-500 focus:outline-none focus:ring-0 focus:border-gray-600 peer disabled:cursor-not-allowed disabled:opacity-50 "
                id="instagramUrl"
                name="instagramUrl"
                type="text"
                value="instagram.com"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="tremor-Flex-root flex w-full flex-row justify-between items-center mt-6"
      >
        <button
          class="tremor-Button-root shrink-0 inline-flex justify-center items-center group font-medium outline-none rounded-tremor-default border shadow-tremor-input dark:shadow-dark-tremor-input text-sm text-tremor-brand-inverted dark:text-dark-tremor-brand-inverted border-tremor-brand dark:border-dark-tremor-brand hover:border-tremor-brand-emphasis dark:hover:border-dark-tremor-brand-emphasis opacity-50 cursor-not-allowed py-3 px-6 rounded-lg border-none bg-gradient-btn-back dark:bg-gradient-btn-back shadow-btn-surface dark:!shadow-btn-surface hover:!shadow-btn-surface-hover dark:hover:!shadow-btn-surface-hover"
          disabled=""
        >
          <span
            class="tremor-Button-text text-tremor-default whitespace-nowrap"
          >
            <p
              class="text-tremor-default uppercase font-bold text-xs text-gray-900 dark:text-black tracking-wide"
            >
              Back
            </p>
          </span>
        </button>
        <button
          class="tremor-Button-root shrink-0 inline-flex justify-center group font-medium outline-none rounded-tremor-default border shadow-tremor-input dark:shadow-dark-tremor-input text-sm text-tremor-brand-inverted border-tremor-brand dark:border-dark-tremor-brand hover:border-tremor-brand-emphasis dark:hover:border-dark-tremor-brand-emphasis opacity-50 cursor-not-allowed rounded-lg py-3 px-6 bg-gradient-primary dark:bg-gradient-pickled hover:dark:!bg-gradient-pickled border-none dark:text-white text-center items-end mt-0"
          disabled=""
          type="submit"
        >
          <span
            class="tremor-Button-text text-tremor-default whitespace-nowrap"
          >
            <p
              class="text-tremor-default uppercase font-bold text-xs text-white dark:text-white tracking-wide"
            >
              Send
            </p>
          </span>
        </button>
      </div>
    </form>
  </div>
</div>
`;
